index.d.ts

```diff
@@ -65,32 +90,26 @@ declare namespace messageControl {
      *
      * `<!--alpha ignore charlie-->
      */
-    known?: string[] | Nullish
-    /**
-     * whether to treat all messages as turned off initially
-     * defeaults to false.
-     */
-    reset?: boolean | Nullish
-    /**
-     * list of `ruleId`s to initially turn on if `reset: true`.
-     */
-    enable?: string[] | Nullish | unknown
-    /** list of `ruleId`s to turn off if `reset: false` */
-    disable?: string[] | Nullish | unknown
```

Comment:
These were moved to the sub-classes, to allow for more fine grained typings

Reply:
honestly thinking about it this is more correct than the approaches I was trying